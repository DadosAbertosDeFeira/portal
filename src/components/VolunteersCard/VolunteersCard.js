import PropTypes from 'prop-types';
import BodyText from '../BodyText';

import Link from '../Link';

import styles from './styles.module.scss';

export default function VolunteersCard({
  name,
  role,
  picture,
  linkedin,
  github,
  instagram,
  twitter,
}) {
  return (
    <div className={styles.volunteers}>
      <img src={picture} alt={name} className={styles.volunteersPicture} />
      <div className={styles.volunteersInfo}>
        <div>
          <BodyText className={styles.volunteersInfoName}>{name}</BodyText>
          <BodyText size="medium" className={styles.volunteersInfoRole}>
            {role}
          </BodyText>
        </div>
        <div className={styles.volunteersIconContainer}>
          {twitter && (
            <Link href={`https://www.twitter.com/${twitter}`}>
              <img src="/assets/icons/twitter.svg" alt="Twitter icon" />
            </Link>
          )}
          {linkedin && (
            <Link href={`https://www.linkedin.com/in/${linkedin}`}>
              <img src="/assets/icons/linkedin.svg" alt="Linkedin icon" />
            </Link>
          )}
          {instagram && (
            <Link href={`https://www.instagram.com/${instagram}`}>
              <img src="/assets/icons/instagram.svg" alt="Instagram icon" />
            </Link>
          )}
          {github && (
            <Link href={`https://github.com/${github}`}>
              <img src="/assets/icons/github.svg" alt="Github icon" />
            </Link>
          )}
        </div>
      </div>
    </div>
  );
}

VolunteersCard.defaultProps = {
  linkedin: null,
  github: null,
  instagram: null,
  twitter: null,
};

VolunteersCard.propTypes = {
  name: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
  linkedin: PropTypes.string,
  github: PropTypes.string,
  instagram: PropTypes.string,
  twitter: PropTypes.string,
};
