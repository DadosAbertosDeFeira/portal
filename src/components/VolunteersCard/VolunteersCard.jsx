import PropTypes from "prop-types";
import Image from "next/image";

import Link from "next/link";
import { Text } from "@/components/atoms/Text";

import Twitter from "../../../public/assets/icons/twitter.svg";
import Instagram from "../../../public/assets/icons/instagram.svg";
import LinkedIn from "../../../public/assets/icons/linkedin.svg";
import Github from "../../../public/assets/icons/github.svg";

export default function VolunteersCard({
  name,
  role,
  picture,
  linkedin,
  github,
  instagram,
  twitter,
}) {
  return (
    <div className="flex flex-row p-4 max-w-[390px] min-w-[288px] w-full rounded-[4px] border-solid border-t-[3px] border-primary-dark bg-gray-lightest shadow">
      <Image
        src={picture}
        alt={name}
        width={70}
        height={70}
        className="bg-white rounded-[50%] mr-4"
      />
      <div className="flex flex-col justify-between">
        <div className="flex flex-col">
          <Text className="">{name}</Text>
          <Text className="mt-[0.1rem] text-primary-dark">{role}</Text>
        </div>
        <div className="flex flex-row">
          {twitter && (
            <Link
              className="mr-4"
              href={`https://www.twitter.com/${twitter}`}
              target="_self"
            >
              <Image className="h-5 w-5" src={Twitter} alt="Twitter icon" />
            </Link>
          )}
          {linkedin && (
            <Link
              className="mr-4"
              href={`https://www.linkedin.com/in/${linkedin}`}
              target="_blank"
              rel="noreferrer"
            >
              <Image className="h-5 w-5" src={LinkedIn} alt="Linkedin icon" />
            </Link>
          )}
          {instagram && (
            <Link
              className="mr-4"
              href={`https://www.instagram.com/${instagram}`}
              target="_blank"
              rel="noreferrer"
            >
              <Image className="h-5 w-5" src={Instagram} alt="Instagram icon" />
            </Link>
          )}
          {github && (
            <Link
              className="mr-4"
              href={`https://github.com/${github}`}
              target="_blank"
              rel="noreferrer"
            >
              <Image className="h-5 w-5" src={Github} alt="Github icon" />
            </Link>
          )}
        </div>
      </div>
    </div>
  );
}

VolunteersCard.defaultProps = {
  linkedin: null,
  github: null,
  instagram: null,
  twitter: null,
};

VolunteersCard.propTypes = {
  name: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
  linkedin: PropTypes.string,
  github: PropTypes.string,
  instagram: PropTypes.string,
  twitter: PropTypes.string,
};
