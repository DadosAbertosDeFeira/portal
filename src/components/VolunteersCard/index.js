import PropTypes from 'prop-types';
import BodyText from '../BodyText';

import Link from '../Link';

import styles from './styles.module.scss';

export default function VolunteersCard({
  name,
  role,
  picture,
  linkedin,
  github,
  instagram,
}) {
  return (
    <div className={styles.volunteers}>
      <img src={picture} alt={name} className={styles.volunteersPicture} />
      <div className={styles.volunteersInfo}>
        <div>
          <BodyText className={styles.volunteersInfoName}>{name}</BodyText>
          <BodyText size="medium" className={styles.volunteersInfoRole}>
            {role}
          </BodyText>
        </div>
        <div className={styles.volunteersIconContainer}>
          {linkedin && (
            <Link href={`https://www.linkedin.com/in/${linkedin}`}>
              <img src="/assets/icons/linkedin.svg" alt="Linkedin icon" />
            </Link>
          )}
          {instagram && (
            <Link href={`https://www.instagram.com/${instagram}`}>
              <img src="/assets/icons/instagram.svg" alt="Instagram icon" />
            </Link>
          )}
          {github && (
            <Link href={`https://github.com/${github}`}>
              <img src="/assets/icons/github.svg" alt="Github icon" />
            </Link>
          )}
        </div>
      </div>
    </div>
  );
}

VolunteersCard.defaultProps = {
  linkedin: null,
  github: null,
  instagram: null,
};

VolunteersCard.propTypes = {
  name: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
  linkedin: PropTypes.string,
  github: PropTypes.string,
  instagram: PropTypes.string,
};
