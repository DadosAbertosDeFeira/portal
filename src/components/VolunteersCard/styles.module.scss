@import '../../styles/variables.scss';

.volunteers {
  display: flex;
  flex-direction: row;

  padding: 1rem;
  max-width: 390px;
  min-width: 288px;
  width: 100%;
  border-radius: 4px;
  border-top: 3px solid $color-primary--dark;
  background: $color-gray-4;
  box-shadow: $shadow-1;

  .volunteersPicture {
    background: $color-white;
    border-radius: 50%;
    margin-right: 1rem;
  }
  .volunteersInfo {
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    .volunteersInfoName strong {
      color: $color-primary--dark;
    }

    .volunteersInfoRole {
      margin-top: 0.1rem;
      color: $color-primary;
    }

    div {
      display: flex;
      flex-direction: column;
    }

    .volunteersIconContainer {
      flex-direction: row;
      a {
        margin-right: 1rem;
        img {
          height: 20px;
          width: 20px;
        }
      }
    }
  }
}
