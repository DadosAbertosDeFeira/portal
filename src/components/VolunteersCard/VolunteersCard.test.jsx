/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render, screen } from "@testing-library/react";
import VolunteersCard from ".";

describe("<VolunteersCard />", () => {
  let sharedProps;

  beforeEach(() => {
    sharedProps = {
      name: "Scooby Doo",
      role: "Detective",
      picture: "https://scoobydoo.com/profile.jpeg",
    };
  });

  it("renders correctly without social network links", () => {
    const { asFragment } = render(<VolunteersCard {...sharedProps} />);

    expect(screen.queryByAltText("Twitter icon")).not.toBeInTheDocument();
    expect(screen.queryByAltText("Linkedin icon")).not.toBeInTheDocument();
    expect(screen.queryByAltText("Instagram icon")).not.toBeInTheDocument();
    expect(screen.queryByAltText("Github icon")).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with social network links", () => {
    const props = {
      ...sharedProps,
      twitter: "scoobydoo",
      github: "scoobydoo",
      instagram: "scoobydoo_detective",
      linkedin: "scoobydoo_engineer",
    };
    const { asFragment } = render(<VolunteersCard {...props} />);

    expect(screen.getByAltText("Twitter icon")).toBeInTheDocument();
    expect(screen.getByAltText("Linkedin icon")).toBeInTheDocument();
    expect(screen.getByAltText("Instagram icon")).toBeInTheDocument();
    expect(screen.getByAltText("Github icon")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
