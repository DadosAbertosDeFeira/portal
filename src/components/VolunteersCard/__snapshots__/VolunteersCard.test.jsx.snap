// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolunteersCard /> renders correctly with social network links 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row p-4 max-w-[390px] min-w-[288px] w-full rounded-[4px] border-solid border-t-[3px] border-primary-dark bg-gray-lightest shadow"
  >
    <img
      alt="Scooby Doo"
      class="bg-white rounded-[50%] mr-4"
      data-nimg="1"
      decoding="async"
      height="70"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fscoobydoo.com%2Fprofile.jpeg&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fscoobydoo.com%2Fprofile.jpeg&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fscoobydoo.com%2Fprofile.jpeg&w=256&q=75 2x"
      style="color: transparent;"
      width="70"
    />
    <div
      class="flex flex-col justify-between"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="text-black text-sm tablet:text-base "
          data-testid=""
        >
          Scooby Doo
        </p>
        <p
          class="text-black text-sm tablet:text-base mt-[0.1rem] text-primary-dark"
          data-testid=""
        >
          Detective
        </p>
      </div>
      <div
        class="flex flex-row"
      >
        <a
          class="mr-4"
          href="https://www.twitter.com/scoobydoo"
          target="_self"
        >
          <img
            alt="Twitter icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <a
          class="mr-4"
          href="https://www.linkedin.com/in/scoobydoo_engineer"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Linkedin icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <a
          class="mr-4"
          href="https://www.instagram.com/scoobydoo_detective"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Instagram icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <a
          class="mr-4"
          href="https://github.com/scoobydoo"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Github icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<VolunteersCard /> renders correctly without social network links 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row p-4 max-w-[390px] min-w-[288px] w-full rounded-[4px] border-solid border-t-[3px] border-primary-dark bg-gray-lightest shadow"
  >
    <img
      alt="Scooby Doo"
      class="bg-white rounded-[50%] mr-4"
      data-nimg="1"
      decoding="async"
      height="70"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fscoobydoo.com%2Fprofile.jpeg&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fscoobydoo.com%2Fprofile.jpeg&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fscoobydoo.com%2Fprofile.jpeg&w=256&q=75 2x"
      style="color: transparent;"
      width="70"
    />
    <div
      class="flex flex-col justify-between"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="text-black text-sm tablet:text-base "
          data-testid=""
        >
          Scooby Doo
        </p>
        <p
          class="text-black text-sm tablet:text-base mt-[0.1rem] text-primary-dark"
          data-testid=""
        >
          Detective
        </p>
      </div>
      <div
        class="flex flex-row"
      />
    </div>
  </div>
</DocumentFragment>
`;
